routers/api/packages/npm/npm.go

```diff
@@ -113,6 +116,26 @@ func UploadPackage(ctx *context.Context) {
 		return
 	}
 
+	repo, err := repo_model.GetRepositoryByURL(npmPackage.Metadata.RepositoryURL)
+	if err == nil {
+		flag := repo.OwnerID == ctx.Doer.ID
+
+		if !flag {
+			perms, err := access_model.GetUserRepoPermission(ctx, repo, ctx.Doer)
+			if err != nil {
+				apiError(ctx, http.StatusInternalServerError, err)
+				return
+			}
+
+			flag = perms.CanWrite(unit.TypePackages)
+		}
+
+		if !flag {
```

Comment:
```suggestion
		canWrite := repo.OwnerID == ctx.Doer.ID

		if !canWrite {
			perms, err := access_model.GetUserRepoPermission(ctx, repo, ctx.Doer)
			if err != nil {
				apiError(ctx, http.StatusInternalServerError, err)
				return
			}

			canWrite = perms.CanWrite(unit.TypePackages)
		}

		if !canWrite {
```

better variable name